rmmxapianapi.cc

```diff
@@ -523,6 +524,58 @@ extern "C" {
       return numfolders;
     }
 
+    // returns a pair: [total, unread] in `results[]`
+    int EMSCRIPTEN_KEEPALIVE getFolderMessageCounts(const char *folderName, int results[]) {
+        if (!dbc) return 0;
+
+        Xapian::QueryParser queryparser;
+        queryparser.set_database(dbc->db);
+        queryparser.add_boolean_prefix("folder", "XFOLDER:");
+        queryparser.add_boolean_prefix("flag", "XF");
+
+        string queryString = "folder:\"";
+        queryString += folderName;
+        queryString += "\"";
```

Comment:
You don't need to build a query string and parse it to create a `Xapian::Query` object, and it's both slower and introduces corner cases (for example, what if `folderName` contains the character `"`?)

You can create the `Xapian::Query` objects for the terms directly via the API, and compose them as you want - e.g.:

```
Xapian::Query query("XFOLDER:" + folderName);
query &= ~Xapian::Query("XFseen");
```

Or you can explicitly construct a parent object specifying `OP_AND_NOT` if you don't like operator overloading.